service/history/historyEngine.go

```diff
@@ -585,44 +585,98 @@ func (e *historyEngineImpl) GetMutableState(
 func (e *historyEngineImpl) QueryWorkflow(
 	ctx ctx.Context,
 	request *h.QueryWorkflowRequest,
-) (*h.QueryWorkflowResponse, error) {
+) (retResp *h.QueryWorkflowResponse, retErr error) {
+	domainCache, err := e.shard.GetDomainCache().GetDomainByID(request.GetDomainUUID())
+	if err != nil {
+		return nil, err
+	}
+	ttl := e.shard.GetConfig().LongPollExpirationInterval(domainCache.GetInfo().Name)
+	timer := time.NewTimer(ttl)
+	defer timer.Stop()
+
 	context, release, err := e.historyCache.getOrCreateWorkflowExecution(ctx, request.GetDomainUUID(), *request.GetExecution())
 	if err != nil {
 		return nil, err
 	}
-	ms, err := context.loadWorkflowExecution()
```

Comment:
plz add a `defer func() { release(retError) }()` here
I know there is a release fn call after error checking and `msBuilder.GetQueryRegistry()`
the defer is more of a best practice

Reply:
Oh thank you - I did it this way because I did not realize it was idempotent.